packages/widget-v2/src/hooks/useGetSourceBalance.ts

```diff
@@ -15,6 +15,12 @@ export const useGetSourceBalance = () => {
     const { chainID, denom } = sourceAsset;
     if (!denom || !chainID) return;
 
+    const denomsExists = !!skipBalances?.chains?.[chainID]?.denoms;
+    const sourceBalance = skipBalances?.chains?.[chainID]?.denoms?.[denom];
+
+    if (denomsExists && sourceBalance === undefined) {
+      return null;
+    }
```

Comment:
this is because when we don't pass in any denoms, the server will not return back the denoms for the balances that are 0